analyzers/tests/SonarAnalyzer.Test/AnalysisContext/SonarAnalysisContextTest.Register.cs

```diff
@@ -266,6 +267,156 @@ public void SonarCompilationStartAnalysisContext_RegisterSymbol_ReportsIssue()
+    [DataRow("MainSourceScope", "//<auto-generated />", "Node", "CodeBlockStart_Node", "CodeBlock", "CodeBlockStart_End")] // Note: "SymbolEnd" is missing here. ReportIssues does not forward the call.
+                                                                                                                           // https://github.com/SonarSource/sonar-dotnet/issues/8876
+    [DataRow("TestSourceScope", "")]
+    [DataRow("TestSourceScope", "//<auto-generated />")]
+    public async Task SonarCompilationStartAnalysisContext_RegisterSymbolStartAction_RegisterAndReporting_ScopeAndGeneratedCode(
+        string scope, string autogenerated, params string[] expectedDiagnostics)
+    {
+        var snippet = new SnippetCompiler($$"""
+            {{autogenerated}}
+            class C
+            {
+                public void M() => ToString();
+            }
+            """);
+        var diagnosticDescriptor = new DiagnosticDescriptor("TEST", "Title", "{0}", "Category", DiagnosticSeverity.Warning, true, customTags: [scope]);
+        var analyzer = new TestAnalyzerCS(diagnosticDescriptor, analysisContext =>
+            analysisContext.RegisterCompilationStartAction(compilationStartContext =>
+                compilationStartContext.RegisterSymbolStartAction(symbolStartContext =>
+                {
+                    symbolStartContext.RegisterCodeBlockAction(codeBlockContext =>
+                    {
+                        codeBlockContext.ReportIssue(CreateDiagnostic("CodeBlock"));
+                    });
+                    symbolStartContext.RegisterCodeBlockStartAction<SyntaxKind>(codeBlockStartContext =>
+                    {
+                        codeBlockStartContext.RegisterNodeAction(nodeContext =>
+                            nodeContext.ReportIssue(CreateDiagnostic("CodeBlockStart_Node")), SyntaxKind.InvocationExpression);
+                        codeBlockStartContext.RegisterCodeBlockEndAction(codeBlockEndContext =>
+                            codeBlockEndContext.ReportIssue(CreateDiagnostic("CodeBlockStart_End")));
+                    });
+                    symbolStartContext.RegisterSymbolEndAction(symbolEndContext =>
+                        symbolEndContext.ReportIssue(CSharpGeneratedCodeRecognizer.Instance, CreateDiagnostic("SymbolEnd")));
+                    symbolStartContext.RegisterSyntaxNodeAction(nodeContext =>
+                        nodeContext.ReportIssue(CreateDiagnostic("Node")), SyntaxKind.InvocationExpression);
+                },
+            SymbolKind.NamedType)));
+        var compilation = snippet.Compilation.WithAnalyzers(ImmutableArray.Create<DiagnosticAnalyzer>(analyzer));
+        var diagnostics = await compilation.GetAllDiagnosticsAsync();
+        diagnostics.Should().HaveCount(expectedDiagnostics.Length);
+        diagnostics.Select(diagnostic => diagnostic.GetMessage()).Should().BeEquivalentTo(expectedDiagnostics);
```

Comment:
Yes, to all three points.
I do not see value in duplicating the test to do the `var orderOfCalls = new List<string>();` check. The action in `RegisterCodeBlockStartAction` must have been called to do the other registrations. It is, therefore, implicitly tested that the call happened because we see reports caused by registrations in there.

Reply:
I added a comment about ordering with a link to the documentation.

Reply:
Excellent: the comment is good!